ValveResourceFormat/IO/GltfModelExporter.cs

```diff
@@ -32,107 +36,260 @@ public class GltfModelExporter
-                                translationDict[bone].Add(time, boneFrame.Value.Position);
-                            }
-                            time += 1 / animation.Fps;
-                        }
+                // TODO: skybox/skydome
 
-                        foreach (var bone in rotationDict.Keys)
-                        {
-                            var node = joints.FirstOrDefault(n => n.Name == bone);
-                            if (node != null)
-                            {
-                                exportedAnimation.CreateRotationChannel(node, rotationDict[bone], true);
-                                exportedAnimation.CreateTranslationChannel(node, translationDict[bone], true);
-                            }
-                        }
+                var model = (VModel)modelResource.DataBlock;
+                var skinName = entity.GetProperty<string>("skin");
+                if (skinName == "0" || skinName == "default")
+                {
+                    skinName = null;
+                }
+
+                var transform = CalculateTransformationMatrix(entity);
+                // Add meshes and their skeletons
+                var meshes = LoadModelMeshes(model, Path.GetFileNameWithoutExtension(modelName));
+                for (var i = 0; i < meshes.Length; i++)
+                {
+                    var meshName = meshes[i].Name;
+                    if (skinName != null)
+                    {
+                        meshName += "." + skinName;
                     }
+                    var node = AddMeshNode(exportedModel, scene, model,
+                        meshName, meshes[i].Mesh, model.GetSkeleton(i),
+                        skinName != null ? GetSkinPathFromModel(model, skinName) : null );
+
+                    if (node == null) continue;
+                    // Swap Rotate upright, scale inches to meters.
+                    node.WorldMatrix = transform * TRANSFORMSOURCETOGLTF;
                 }
-                else
+            }
+
+            foreach (var childEntityName in entityLump.GetChildEntityNames())
+            {
+                if (childEntityName == null) continue;
+                var childEntityLumpResource = FileLoader.LoadFile(childEntityName + "_c");
+                if (childEntityLumpResource == null) continue;
+
+                var childEntityLump = (VEntityLump)childEntityLumpResource.DataBlock;
+                LoadEntityMeshes(exportedModel, scene, childEntityLump);
+            }
+        }
+
+        private static string GetSkinPathFromModel(VModel model, string skinName)
+        {
+            var materialGroupForSkin = model.Data.GetArray<IKeyValueCollection>("m_materialGroups")
+                .ToList()
+                .Where(m => m.GetProperty<string>("m_name") == skinName)
+                .SingleOrDefault();
+
+            if (materialGroupForSkin == null) return null;
```

Comment:
There's a few places where you do this, but please use brackets like so:

```csharp
if (materialGroupForSkin == null)
{
    return null;
}
```

*Need to add editorconfig so this is enforced*